app/controllers/SummaryController.scala

```diff
@@ -53,9 +53,13 @@ class SummaryController @Inject()(
     val typeOfJourney = request.choice.value
 
     cacheMapOpt.map {
-      case Some(data) if typeOfJourney == Arrival   => Ok(arrivalSummaryPage(data))
-      case Some(data) if typeOfJourney == Departure => Ok(departureSummaryPage(data))
-      case _                                        => handleError("Could not obtain data from DB")
+      case Some(data) if typeOfJourney == Arrival =>
+        Ok(arrivalSummaryPage(data))
+      case Some(data) if typeOfJourney == Departure =>
+        Ok(departureSummaryPage(data))
+      case _ =>
+        logger.warn(s"No movement data found in cache.")
+        errorHandler.getInternalServerErrorPage
```

Comment:
Prefer previous version, but of course with changed method and using `errorHandler.getInternalServerErrorPage` instead of 
```
InternalServerError(
      errorHandler.standardErrorTemplate(
        pageTitle = Messages("global.error.title"),
        heading = Messages("global.error.heading"),
        message = Messages("global.error.message")
      )
    )
```

So in general I mean it's better to have it in small method

Reply:
Maybe it should be move to local action Refiner ?